core/services/job/models.go

```diff
@@ -893,13 +895,13 @@ const (
 )
 
 // Validate checks the workflow spec for correctness
-func (w *WorkflowSpec) Validate(ctx context.Context) error {
+func (w *WorkflowSpec) Validate(ctx context.Context, logger logger.Logger) error {
 	s, err := pkgworkflows.ParseWorkflowSpecYaml(w.Workflow)
 	if err != nil {
 		return fmt.Errorf("%w: failed to parse workflow spec %s: %w", ErrInvalidWorkflowYAMLSpec, w.Workflow, err)
 	}
 
-	if _, err = w.SDKSpec(ctx); err != nil {
+	if _, err = w.SDKSpec(ctx, logger); err != nil {
```

Comment:
Oh I think I understand the need for the logger, but if we throw away the returned Spec here, could we just pass a no-op logger instead?

Reply:
I shouldn't have been trying to parse it as YAML above, that explains why I didn't need the spec.  I'll add a test for this.